modules/openapi-generator-online/src/main/java/org/openapitools/codegen/online/configuration/OpenAPIDocumentationConfig.java

```diff
@@ -74,6 +81,27 @@ public Docket customImplementation(){
                 .ignoredParameterTypes(Resource.class)
                 .ignoredParameterTypes(InputStream.class)
                 .apiInfo(apiInfo());
+
+        String hostString = System.getenv("GENERATOR_HOST");
+        if (!StringUtils.isBlank(hostString)) {
+            try {
+                URI hostURI = new URI(hostString);
+                String scheme = hostURI.getScheme();
+                if (scheme != null) {
+                    docket.protocols(Set.of(new String[] { scheme }));
+                }
+                String authority = hostURI.getAuthority();
+                if (authority != null) {
+                    // In Swagger host refers to host _and_ port, a.k.a. the URI authority
```

Comment:
@fgreinacher thanks for the PR. Can you please replace `Swagger` with `OpenAPI`?

Reply:
@wing328 Sure, done!

I also pushed https://github.com/OpenAPITools/openapi-generator/pull/12064/commits/1e3eab6e413ee4978590403372015e68590f2365 to make the new code compatible with Java 8 used in CI.

Reply:
Sorry, needed another fix: https://github.com/OpenAPITools/openapi-generator/pull/12064/commits/36af1476d95de63f64f9679dbca7a3e2baa0ecd7

Reply:
@fgreinacher I tested with the following:

```
GENERATOR_HOST=123 java -jar ./target/openapi-generator-online.jar
``` 

but I don't see any warning message from the following line:
```
                LOGGER.warn("Could not parse configured GENERATOR_HOST '" + hostString + "': " + e.getMessage());
```

output:
```
➜  openapi-generator-online git:(fgreinacher-fix/generator-host-swagger-docs) GENERATOR_HOST=123 java -jar ./target/openapi-generator-online.jar
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/williamcheng/Code/openapi-generator4/modules/openapi-generator-online/target/openapi-generator-online.jar!/BOOT-INF/lib/logback-classic-1.2.6.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/williamcheng/Code/openapi-generator4/modules/openapi-generator-online/target/openapi-generator-online.jar!/BOOT-INF/lib/slf4j-simple-1.7.32.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v2.5.5)

2022-04-09 09:49:35.032  INFO 2797 --- [           main] o.o.codegen.online.OpenAPI2SpringBoot    : Starting OpenAPI2SpringBoot v6.0.0-SNAPSHOT using Java 11.0.13 on WILLIAMs-MacBook-Pro.local with PID 2797 (/Users/williamcheng/Code/openapi-generator4/modules/openapi-generator-online/target/openapi-generator-online.jar started by williamcheng in /Users/williamcheng/Code/openapi-generator4/modules/openapi-generator-online)
2022-04-09 09:49:35.038  INFO 2797 --- [           main] o.o.codegen.online.OpenAPI2SpringBoot    : No active profile set, falling back to default profiles: default
2022-04-09 09:49:38.295  INFO 2797 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2022-04-09 09:49:38.320  INFO 2797 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2022-04-09 09:49:38.321  INFO 2797 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.53]
2022-04-09 09:49:38.537  INFO 2797 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2022-04-09 09:49:38.538  INFO 2797 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 3303 ms
2022-04-09 09:49:40.663  INFO 2797 --- [           main] o.s.b.a.w.s.WelcomePageHandlerMapping    : Adding welcome page: class path resource [static/index.html]
2022-04-09 09:49:40.824  WARN 2797 --- [           main] o.s.b.a.m.MustacheAutoConfiguration      : Cannot find template location: classpath:/templates/ (please add some templates, check your Mustache configuration, or set spring.mustache.check-template-location=false)
2022-04-09 09:49:40.987  INFO 2797 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2022-04-09 09:49:41.614  INFO 2797 --- [           main] o.o.codegen.online.OpenAPI2SpringBoot    : Started OpenAPI2SpringBoot in 13.077 seconds (JVM running for 14.373)

```

Am I testing the change correctly? How did you test locally?

Reply:
@wing328 `123` is actually a valid (relative) URI. You need to use invalid characters, e.g. `GENERATOR_HOST='\abc' java -jar ./target/openapi-generator-online.jar`